Jenkinsfile

```diff
@@ -189,11 +189,7 @@ stage('Publish') {
                     """
                     sh "scripts/create-github-release.js"
 
-                    // Publish the updated package to the npm registry.
-                    // Use `npm` rather than `yarn` for publishing.
-                    // See https://github.com/yarnpkg/yarn/pull/3391.
-                    sh "echo '//registry.npmjs.org/:_authToken=${env.NPM_TOKEN}' >> \$HOME/.npmrc"
-                    sh "npm publish --tag ${npmTag}"
+                    sh "NPM_TOKEN=${env.NPM_TOKEN} yarn publish --no-interactive --tag ${npmTag} --new-version=${newPkgVersion}"
```

Comment:
Have you confirmed that `yarn publish` definitely uses `NPM_TOKEN` if it is set this way and `.npmrc` is not present?

From what I can tell looking at the yarn issues there is no built-in support for `NPM_TOKEN` in yarn itself. Instead there is a feature where `.npmrc` can contain references to environment variables and there is a convention to write `//registry.npmjs.org/:_authToken=${NPM_TOKEN}` into the `.npmrc` file. See https://github.com/yarnpkg/yarn/pull/1207.

I did however find a reference to `NPM_AUTH_TOKEN` (and also `YARN_AUTH_TOKEN`) in the yarn sources: https://github.com/yarnpkg/yarn/blob/a4708b29ac74df97bac45365cba4f1d62537ceb7/src/cli/commands/login.js#L64

The other issue I see here is that a `.npmrc` file could currently be left over from a previous build and I'm not sure what happens if the auth token is set in both places.

The simplest remedy I can see here is to keep the "sh echo '..." line unless you're sure that setting `NPM_TOKEN` on its own works, and just change the line that invokes `npm publish`.

Reply:
Yes, I confirm that it works with NPM_TOKEN.

I have reverted the `sh "echo...` line.